### DISCUSSION TITLE
Cannot get same performance as kong load test

### DISCUSSION THREAD

**Discussion by alexeiba** [Kong/kong#8411] [Help]
Hello , Im trying to test the performance of simple proxy with kong single core. 
![image](https://user-images.githubusercontent.com/97094698/153901673-f64bdff3-8b33-443c-aee8-3f61d4e83425.png)

-- To Kong 

Running 5s test @ http://10.21.0.52:8000
  4 threads and 512 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   138.13ms  235.12ms   1.04s    82.80%
    Req/Sec     7.66k     2.24k   13.19k    69.00%
  152405 requests in 5.05s, 27.17MB read
  Socket errors: connect 0, read 409, write 0, timeout 0
Requests/sec:  30191.67
Transfer/sec:      5.38MB
-- To service directly

Running 5s test @ http://10.21.0.69:8080
  4 threads and 512 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     5.17ms    1.62ms  21.22ms   73.59%
    Req/Sec    24.56k     1.69k   31.58k    87.00%
  489275 requests in 5.05s, 43.39MB read
Requests/sec:  96812.48
Transfer/sec:      8.59MB

you can see that its only 30K .

here is my kong.conf 
```console
nginx_main_worker_processes = 1
nginx_main_worker_rlimit_nofile = NONE
nginx_events_worker_connections = NONE
nginx_events_multi_accept = off
```
here is my kong.yml
```yaml
_format_version: "2.1"

_transform: true
services:
- name: example-service
  url: http://10.21.0.69:8080
  routes:
    - paths:
      - "/"
```

the results i get are different from what you get in your tests .
am i missing something? 
thank you

**Comment by fffonion** [comment]
There are couple of places you can tune on Kong side and upstream side, you can replicate our setup at https://github.com/Kong/kong/blob/master/spec/helpers/perf/drivers/terraform.lua, most notably: ulimit of OS, local port range, CPU scaling governer. The actual number also differs from your upstream/service performance and size of payload.
Also note the error log from kong or upstream to see if there's any resouce exhaust message (open file limit, unreleased sockets etc).
